src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLogger.cs

```diff
@@ -49,286 +46,23 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
-                timeStamp: timestamp,
-                levelString: logLevelString,
-                levelBackground: logLevelColors.Background,
-                levelForeground: logLevelColors.Foreground,
-                messageColor: null,
-                logAsError: logLevel >= Options.LogToStandardErrorThreshold
-            );
-        }
-
-        private LogMessageEntry CreateSystemdLogMessage(StringBuilder logBuilder, LogLevel logLevel, string logName, int eventId, string message, Exception exception)
-        {
-            // systemd reads messages from standard out line-by-line in a '<pri>message' format.
-            // newline characters are treated as message delimiters, so we must replace them.
-            // Messages longer than the journal LineMax setting (default: 48KB) are cropped.
-            // Example:
-            // <6>ConsoleApp.Program[10] Request received
-
-            // loglevel
-            string logLevelString = GetSyslogSeverityString(logLevel);
-            logBuilder.Append(logLevelString);
-
-            // timestamp
-            string timestampFormat = Options.TimestampFormat;
-            if (timestampFormat != null)
-            {
-                DateTime dateTime = GetCurrentDateTime();
-                logBuilder.Append(dateTime.ToString(timestampFormat));
-            }
-
-            // category and event id
-            logBuilder.Append(logName);
-            logBuilder.Append('[');
-            logBuilder.Append(eventId);
-            logBuilder.Append(']');
-
-            // scope information
-            GetScopeInformation(logBuilder, multiLine: false);
-
-            // message
-            if (!string.IsNullOrEmpty(message))
+            if (_stringWriter == null)
             {
-                logBuilder.Append(' ');
-                // message
-                AppendAndReplaceNewLine(logBuilder, message);
+                _stringWriter = new StringWriter();
             }
 
-            // exception
-            // System.InvalidOperationException at Namespace.Class.Function() in File:line X
-            if (exception != null)
-            {
-                logBuilder.Append(' ');
-                AppendAndReplaceNewLine(logBuilder, exception.ToString());
-            }
-
-            // newline delimiter
-            logBuilder.Append(Environment.NewLine);
-
-            return new LogMessageEntry(
-                message: logBuilder.ToString(),
-                logAsError: logLevel >= Options.LogToStandardErrorThreshold
-            );
-
-            static void AppendAndReplaceNewLine(StringBuilder sb, string message)
+            if (!string.IsNullOrEmpty(message) || exception != null)
             {
-                int len = sb.Length;
-                sb.Append(message);
-                sb.Replace(Environment.NewLine, " ", len, message.Length);
+                Formatter.Write(logLevel, _name, eventId.Id, state, exception, formatter, ScopeProvider, _stringWriter);
```

Comment:
>Formatter [](start = 16, length = 9)

should the `message` be logged too?

Reply:
message would already be retrievable using Formatter delegate, State and Exception. which are all parts of LogEntry being passed in.